calendar/src/Components/App.tsx

```diff
@@ -0,0 +1,81 @@
+import { Fragment, useState } from "react";
+import { LocaleType } from "../Utils/dateManipulation";
+import { Header } from "./Header/header";
+import "./main.css";
+import MonthView from "./MonthView/MonthView";
+import { DateFormatter, createDateFormatter } from "../Utils/dateFormatter";
+import { FormData, deleteEvent, getEvents, saveEvent } from "../Utils/database";
+import Modal from "./Modal/Modal";
+import WeekView from "./WeekView/WeekView";
+
+function App() {
+	const [modalBody, setModalBody] = useState<JSX.Element>(<Fragment />);
+	const [isModalVisible, setIsModalVisible] = useState<Boolean>(false);
+	const [dateFormatter, setDateFormatter] = useState<DateFormatter>(
+		createDateFormatter(LocaleType.US)
+	);
+	const [selectedDate, setSelectedDate] = useState<Date>(new Date());
+	const [events, setEvents] = useState<FormData[]>(getEvents());
+
+	const onLocaleChange = (newLocale: LocaleType) => {
+		setDateFormatter(createDateFormatter(newLocale));
+	};
+	const onSelectedDateChange = (newDate: Date) => {
+		setSelectedDate(newDate);
+	};
+	const saveToLocalStorage = (event: FormData) => {
+		const newState = [...events, event];
+		setEvents(newState);
+		saveEvent(event);
+	};
+	const removeFromLocalStorage = (eventId: string) => {
+		setEvents(events.filter((storedEvent) => storedEvent.id !== eventId));
+		deleteEvent(eventId);
+	};
+	const openModal = (children: JSX.Element) => {
```

Comment:
I see you pass `openModal` and `hideModal` methods to many components in the app so you can later pass them down to some button that invokes these methods. Instead of passing them throughout your application, try to create a context and a custom hook.
For example `useModal`, where inside you would keep track of the modal state and export `open` and `hide` methods. Then you will only need to use the hook inside the component that needs to call these methods without passing them down throughout the app.

Reply:
Something like this:
https://upmostly.com/tutorials/modal-components-react-custom-hooks